Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/calibration/Calibration.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.randomcutforest.parkservices.calibration;
+
+public enum Calibration {
+
+    NONE,
+
+    /**
+     * a basic staring point where the itervals are adjusted to be the minimal
+     * necessary based on past error the intervals are smaller -- but the interval
+     * precision will likely be close to 1 - 2 * percentile
+     */
+    MINIMAL,
+
+    /**
+     * a Markov inequality based interval, where the past error and model errors are
+     * additive. The interval precision is likely higher than MINIMAL but so are the
+     * interval.
+     */
+    SIMPLE;
```

Comment:
The difference between Calibration simple and minimal mode

```
void adjust(RangeVector rangeVector, RangeVector other) {
        checkArgument(other.values.length == rangeVector.values.length, " mismatch in lengths");
        for (int i = 0; i < rangeVector.values.length; i++) {
            rangeVector.values[i] += other.values[i];
            rangeVector.upper[i] = max(rangeVector.values[i], rangeVector.upper[i] + other.upper[i]);
            rangeVector.lower[i] = min(rangeVector.values[i], rangeVector.lower[i] + other.lower[i]);
        }
    }

    void adjustMinimal(RangeVector rangeVector, RangeVector other) {
        checkArgument(other.values.length == rangeVector.values.length, " mismatch in lengths");
        for (int i = 0; i < rangeVector.values.length; i++) {
            float oldVal = rangeVector.values[i];
            rangeVector.values[i] += other.values[i];
            rangeVector.upper[i] = max(rangeVector.values[i], oldVal + other.upper[i]);
            rangeVector.lower[i] = min(rangeVector.values[i], oldVal + other.lower[i]);
        }
    }
```

Why can simple mode  deal with stationary data (mean is constant) and minimum model can deal with data with distribution shift (e.g., mean changes)? From the above code, the only difference is how we compute upper and lower. In Simple mode, the upper and lower range adds new value while minimal adds old value.

Reply:
The simple mode adds the error intervals (uncertainty from the model and the actual error distribution). The minimal mode fixes the mean,  but uses the error of "other" (which is the error distribution) and ignores the ranges produced by the model. The choice depends on the assumption on the data, the first is a pessimistic choice -- the second is a minimal choice (just adjusting based on the previous errors).  Using simple may be simpler, but will give wider intervals. There can be many more calibration steps based on assumptions one is willing to make on the input data.